pkg/generate/app/pipeline.go

```diff
@@ -64,7 +64,7 @@ func (pb *pipelineBuilder) NewBuildPipeline(from string, resolvedMatch *Componen
 			return nil, fmt.Errorf("can't build %q: %v", from, err)
 		}
 		input = inputImage
-		if !input.AsImageStream {
+		if !input.AsImageStream && resolvedMatch.Value != "scratch" {
```

Comment:
seems excessive unless we have at least 2 of these conditions.  right now, scratch is an extremely special case, it's not a member of a general class of image types.

so yes, we could do what you suggest but it feels like overkill at the moment.